src/main/kotlin/com/exactpro/th2/check1/rule/AbstractCheckTask.kt

```diff
@@ -126,6 +126,8 @@ abstract class AbstractCheckTask(
     private var hasMessagesInTimeoutInterval: Boolean = false
     private var bufferContainsStartMessage: Boolean = false
     private var isDefaultSequence: Boolean = false
+    protected var isCheckpointLastReceivedMessage = bufferContainsStartMessage && !hasMessagesInTimeoutInterval
```

Comment:
I don't see any places where you set this property a new value. So it is always `false`